src/test/java/uk/nhs/digital/nhsconnect/lab/results/inbound/InboundMessageHandlerTest.java

```diff
@@ -47,56 +60,118 @@
+            () -> verify(meshOutboundQueueService).publish(outboundMeshMessage)
+        );
+    }
+
+    @Test
+    void handleInvalidMessageMeshMessageRaisesException() throws InterchangeParsingException, MessagesParsingException {
+        final MeshMessage meshMessage = new MeshMessage()
+            .setWorkflowId(WorkflowId.PATHOLOGY);
+        when(edifactParser.parse(meshMessage.getContent())).thenThrow(MessagesParsingException.class);
+
+        final OutboundMeshMessage outboundMeshMessage = new MeshMessage()
+            .setWorkflowId(WorkflowId.PATHOLOGY_ACK);
+        when(outboundMeshMessageBuilder.buildNhsAck(eq(WorkflowId.PATHOLOGY), any(MessagesParsingException.class)))
+            .thenReturn(outboundMeshMessage);
+
+        inboundMessageHandler.handle(meshMessage);
+
+        assertAll(
+            () -> verify(edifactParser).parse(any()),
+            () -> verify(edifactToFhirService, never()).convertToFhir(any(Message.class)),
+            () -> verify(gpOutboundQueueService, never()).publish(any(Bundle.class)),
+            () -> verify(outboundMeshMessageBuilder, never()).buildNhsAck(any(), any(), anyList()),
+            () -> verify(meshOutboundQueueService).publish(outboundMeshMessage)
+        );
+    }
+
+    @Test
+    void handleInboundMeshMessageNoMessagesDoesNotPublishToGpOutboundQueue()
+            throws InterchangeParsingException, MessagesParsingException {
+
+        final MeshMessage meshMessage = new MeshMessage()
+            .setWorkflowId(WorkflowId.PATHOLOGY);
         when(edifactParser.parse(meshMessage.getContent())).thenReturn(interchange);
 
+        final OutboundMeshMessage outboundMeshMessage = new MeshMessage()
+            .setWorkflowId(WorkflowId.PATHOLOGY_ACK);
+        when(outboundMeshMessageBuilder
+            .buildNhsAck(eq(WorkflowId.PATHOLOGY), eq(interchange), eq(Collections.emptyList())))
+            .thenReturn(outboundMeshMessage);
+
         inboundMessageHandler.handle(meshMessage);
 
-        //TODO: NIAD-1063 temporarily disabling NHSACK for v0.1
-        //verify(edifactParser, times(2)).parse(any());
-        verify(edifactParser, times(1)).parse(any());
-        verify(edifactToFhirService, never()).convertToFhir(any(Message.class));
-        verify(gpOutboundQueueService, never()).publish(any(Bundle.class));
-        //TODO: NIAD-1063 temporarily disabling NHSACK for v0.1
-        //verify(recepProducerService).produceRecep(interchange);
-        //verify(meshOutboundQueueService).publish(any(OutboundMeshMessage.class));
+        assertAll(
+            () -> verify(edifactParser).parse(any()),
+            () -> verify(edifactToFhirService, never()).convertToFhir(any(Message.class)),
+            () -> verify(gpOutboundQueueService, never()).publish(any(Bundle.class)),
+            () -> verify(outboundMeshMessageBuilder)
+                .buildNhsAck(eq(WorkflowId.PATHOLOGY), eq(interchange), eq(Collections.emptyList())),
```

Comment:
Good example of a case where `eq()` isn't needed. All of the parameters are explicit; but this still works as it is so a change is not necessary - just an FYI

```suggestion
                .buildNhsAck(WorkflowId.PATHOLOGY, interchange, Collections.emptyList()),
```

Reply:
I knew that. Must have been set by accident during refactor